realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -235,43 +246,80 @@ void BaseRealSenseNode::registerDynamicOption(ros::NodeHandle& nh, rs2::options
+            {
+                sensor.set_option(option, option_value);
+            }
+            ddynrec->add(new DDBool(option_name, i, sensor.get_option_description(option), option_value));
             continue;
         }
         std::map<std::string, int> enum_dict = get_enum_method(sensor, option);
         if (enum_dict.empty())
         {
             rs2::option_range op_range = sensor.get_option_range(option);
+            const auto sensor_option_value = sensor.get_option(option);
+            auto option_value = sensor_option_value;
+            if (nh1.param(option_name, option_value, option_value))
+            {
+                if (option_value < op_range.min || op_range.max < option_value)
+                {
+                    ROS_WARN_STREAM("Param '" << nh1.resolveName(option_name) << "' has value " << option
+                            << " outside the range [" << op_range.min << ", " << op_range.max
+                            << "]. Using current sensor value " << sensor_option_value << " instead.");
+                    option_value = sensor_option_value;
+                }
+                else
+                {
+                    sensor.set_option(option, option_value);
+                }
+            }
             if (is_int_option(sensor, option))
             {
-                ddynrec->add(new DDInt(option_name, i, sensor.get_option_description(option), sensor.get_option(option), op_range.min, op_range.max));
+                ddynrec->add(new DDInt(option_name, i, sensor.get_option_description(option), option_value, op_range.min, op_range.max));
             }
             else
             {
-                ddynrec->add(new DDDouble(option_name, i, sensor.get_option_description(option), sensor.get_option(option), op_range.min, op_range.max));
+                ddynrec->add(new DDDouble(option_name, i, sensor.get_option_description(option), option_value, op_range.min, op_range.max));
             }
         }
         else
         {
-            if (int(sensor.get_option(option)) > (int)enum_dict.size())
+            const auto sensor_option_value = sensor.get_option(option);
+            auto option_value = int(sensor_option_value);
+            if (nh1.param(option_name, option_value, option_value))
+            {
+                if (option_value > (int)enum_dict.size())
+                {
+                  ROS_WARN_STREAM("Param '" << nh1.resolveName(option_name) << "' has value " << option << " > "
+                                            << enum_dict.size() << ". Using current sensor value "
+                                            << sensor_option_value << " instead.");
+                  option_value = sensor_option_value;
+                }
+                else
+                {
+                    sensor.set_option(option, option_value);
+                }
+            }
+            if (option_value > (int)enum_dict.size())
```

Comment:
This check does not seem to be needed here (the only possibility of `option_value` to be out of range would be by reading it from the param server - and that already has the check above).

Reply:
@doronhi Added https://github.com/intel-ros/realsense/commit/ccaa46462a02e63033276586b20a71a9188f3fa0#diff-15761e8f7813869ae417a96398deda78R249 in ccaa46462a02e63033276586b20a71a9188f3fa0 , although the commit message doesn't say why. Could you clarify why that check is needed, please?

Reply:
I've removed this validation check, since the option value retrieved from the sensor should always be correct IMHO.